lib/dry/cli/parser.rb

```diff
@@ -30,22 +30,16 @@ def self.call(command, arguments, names)
         end.parse!(arguments)
 
         parsed_options = command.default_params.merge(parsed_options)
-        parse_required_params(command, arguments, names, parsed_options)
+        parse_required_params(command, arguments, prog_name, parsed_options)
       rescue ::OptionParser::ParseError
-        Result.failure("Error: \"#{names.last}\" was called with arguments \"#{original_arguments.join(' ')}\"") # rubocop:disable Metrics/LineLength
-      end
-
-      # @since 0.1.0
-      # @api private
-      def self.full_command_name(names)
-        ProgramName.call(names)
+        Result.failure("ERROR: \"#{prog_name}\" was called with arguments \"#{original_arguments.join(' ')}\"") # rubocop:disable Metrics/LineLength
```

Comment:
@IvanShamatov Never trust user input 😉 . Original `original_arguments.join(' ')` should be passed to `Shellescape.shellescape` or `Shellescape.shelljoin(original_arguments)`.

Reply:
I've never heard of this before. I mean I understand the concept of injection. So I found a couple of references https://www.reddit.com/r/ruby/comments/9lr1mk/what_is_the_use_case_for_shellwordsescape/, 
https://pypi.org/project/shellescape/
And it is said that escaping is useful when you pass results to `system/exec`  , and there is nothing bad with just printing it, even if there is an `rm -rf /`